src/analyze/analyze-verify.c

```diff
@@ -233,6 +253,16 @@ int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run
         if (strv_isempty(filenames))
                 return 0;
 
+        /* Allow systemd-analyze to hook in a callback function so that it can get
+         * all the required log data from the function itself without having to rely
+         * on a global set variable for the same */
+        if (recursive_errors >= 0) {
```

Comment:
so none of our callers passes recursive_errors < 0 here, do we really want to suppor that?